### PR TITLE
Logic for opening example projects

### PR DESCRIPTION
**Pull request by shaal**
## Changes

When opening this Astro repo in Gitpod, an example project would open in the following logic:
* If Gitpod workspace was opened from a specific example project - that project would open
* If $ASTRO_NEW environment variable was set - open that project
* Otherwise, be default, open `starter` example project

## Testing

* Go to https://github.com/shaal/astro/tree/gitpod--open-examples/examples
(that's a fork of Astro on my user account, with the branch that is the base of this PR. It will include the information of which directory Gitpod workspace was opened from
* Click on any example project.
* Either use Gitpod browser extension, or add `https://gitpod.io/#` before the full URL, for example -
https://gitpod.io/#https://github.com/shaal/astro/tree/gitpod--open-examples/examples/framework-lit
* Confirm that the correct example project opened.

Confirm that ASTRO_NEW is still working, by adding
https://gitpod.io/#ASTRO_NEW=blog/https://github.com/shaal/astro/tree/gitpod--open-examples

Confirm that `starter` project starting when just opening the branch without any additions -
https://gitpod.io/#https://github.com/shaal/astro/tree/gitpod--open-examples

### CODE REVIEW

**Review by natemoo-re — APPROVED** [review]
This looks like a clean refactor—approved! Let me know if there are any more changes you want to land before I merge.

**Comment by shaal** [comment]
Thank you @natemoo-re 
These are all the changes I had in mind.

**Pull request merged**
